Xamarin.Forms.Platform.WPF/Renderers/MediaElementRenderer.cs

```diff
@@ -0,0 +1,280 @@
+					}
+					else if (Element.Source.LocalPath.StartsWith("/temp"))
+					{
+						filePath = System.IO.Path.Combine(System.IO.Path.GetTempPath(), Element.Source.LocalPath.Substring(6));
+					}
+					else
+					{
+						throw new ArgumentException("Invalid Uri", "Source");
+					}
+
+					Control.Source = new Uri(filePath);
+				}
+				else if (Element.Source.Scheme == "https")
+				{
+					throw new ArgumentException("HTTPS Not supported", "Source");
+				}
+				else
+				{
+					Control.Source = Element.Source;
+				}
+
+				Controller.CurrentState = MediaElementState.Opening;
+			}
+		}
+
+		void Control_BufferingEnded(object sender, RoutedEventArgs e)
+		{
+			Controller.BufferingProgress = 1.0;
+			if (Element.AutoPlay)
+			{
+				Controller.CurrentState = MediaElementState.Playing;
+			}
+			else
+			{
+				Controller.CurrentState = MediaElementState.Paused;
+			}
+		}
+
+		void Control_BufferingStarted(object sender, RoutedEventArgs e)
+		{
+			Controller.BufferingProgress = 0.0;
+			Controller.CurrentState = MediaElementState.Buffering;
+		}
+
+		void Control_MediaFailed(object sender, ExceptionRoutedEventArgs e)
+		{
+			Element.OnMediaFailed();
+		}
+
+		void Control_MediaEnded(object sender, RoutedEventArgs e)
+		{
+			if(Element.IsLooping)
+			{
+				// restart media
+				Control.Position = TimeSpan.Zero;
+				Control.Play();
+			}
+			else
+			{
+				Element.OnMediaEnded();
+			}
+		}
+
+		void Control_MediaOpened(object sender, RoutedEventArgs e)
+		{
+			Controller.Duration = Control.NaturalDuration.HasTimeSpan ? Control.NaturalDuration.TimeSpan : (TimeSpan?)null;
+			Controller.VideoHeight = Control.NaturalVideoHeight;
+			Controller.VideoWidth = Control.NaturalVideoWidth;
+
+			Element?.RaiseMediaOpened();
+
+			if(Element.AutoPlay)
+			{
+				Control.Play();
+				Controller.CurrentState = MediaElementState.Playing;
+			}
+		}
+		
+		void Control_SeekCompleted(object sender, RoutedEventArgs e)
+		{
+			Element?.RaiseSeekCompleted();
+		}
+
+		protected override void OnElementPropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+		{
+			switch (e.PropertyName)
+			{
+				//TODO: AreTransportControlsEnabled not supported for WPF
+
+				case nameof(MediaElement.Aspect):
+					Control.Stretch = Element.Aspect.ToStretch();
+					break;
+					
+				case nameof(MediaElement.KeepScreenOn):
+					if (Element.KeepScreenOn)
+					{
+						if (Element.CurrentState == MediaElementState.Playing)
+						{
+							DisplayRequestActive();
+						}
+					}
+					else
+					{
+						DisplayRequestRelease();
+					}
+					break;
+
+				case nameof(MediaElement.Source):
+					UpdateSource();
+					break;
+			}
+
+			base.OnElementPropertyChanged(sender, e);
+		}
+
+		protected override void UpdateWidth()
+		{
+			if (Element.Width > 0)
+			{
+				Control.Width = Element.Width;
+			}
+		}
+
+		protected override void UpdateHeight()
+		{
+			if (Element.Height > 0)
+			{
+				Control.Height = Element.Height;
+			}
```

Comment:
if `Element.Height` is negative value then `Control.Height` nothing will change?
i think better always changes `Height` but with limits. `Control.Height = Math.Max(0, Element.Height)`
also for Width.